src/osd/scrubber/scrub_machine.cc

```diff
@@ -257,15 +259,17 @@ ReservingReplicas::~ReservingReplicas()
   // it's OK to try and clear the flag even if we don't hold it
   // (the flag remembers the actual holder)
   scrbr->clear_reserving_now();
+  // ignore any late-arriving reservation replies
+  context<PrimaryActive>().last_request_sent_nonce++;
```

Comment:
This already gets incremented in ReplicaReservations::send_next_reservation_or_complete().  Either strategy could actually work:
- doing it here would mean that a nonce is associated with a "round" of reservations and is incremented once complete or failed.
- doing it in ReplicaReservations::send_next_reservation_or_complete() would mean that a nonce is associated with a single request.

The latter gets my vote due to not needing to explain a "round", and the comment on last_request_sent_nonce indicates "reservation request is therefore associated with a nonce incremented within an interval with each reservation request".  Either way, let's not not do it in both places -- it'll be confusing.

Reply:
@athanatos,
I will remove it. It was added to serve as an additional guard before "harsh reactions" to
requests arriving from the wrong source:

both in handle_reserve_rejection() (scrub_reservations.cc:259) and in handle_reserve_grant()
(scrub_reservations.cc:169) we first check the nonce before checking (and possibly warning/aborting)
the source. The added increment was added to guarantee that for stale messages (from a previous
round altogether) we would not get a false positive.

As I do not see a sequence that would cause that, I will remove the extra increment.